rules-tests/PHPUnit/Rector/Class_/RemoveDataProviderTestPrefixRector/Fixture/base_class.php.inc

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Rector\Tests\PHPUnit\Rector\Class_\RemoveDataProviderTestPrefixRector\Fixture;
+
+class BaseClass extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @dataProvider testProvideData()
+     */
+    public function test()
+    {
+        $nothing = 5;
+    }
+
+}
+
+namespace Rector\Tests\PHPUnit\Rector\Class_\RemoveDataProviderTestPrefixRector\Fixture;
```

Comment:
This would crash, it's not possible to have 2 namespaces in one file.

Also the rule cannot reach the child classes, so this case is not possible.

How did you got to a situation when you provide data from a child class? :) That seems like wrong way to use them.

Reply:
> How did you got to a situation when you provide data from a child class?

It's more like the base class provides the test method to the child class. The base class itself is abstract in my case. Each of the child classes call the same test method what with difference data in their data providers.

Reply:
In that case you have 2 options:

* refactor to same level
* ignore in rector.php this file and rule combination

Reply:
Sounds good, @TomasVotruba !

Thanks for the help. Cheers.

Reply:
Glad to help 👍